<!-- -->
<launch>

	<arg name="robot_name" default="social_robot"/>
    <arg name="use_rviz" default="true"/>
    <arg name="use_knowledge" default="true"/>
    <arg name="use_reasoner" default="true"/>
    <arg name="use_behavior" default="true"/>
    <arg name="sim" default="true"/>
    <arg name="sim_env" default="default"/>
    <arg name="sync_mode" default="false"/>
    <arg name="use_mongodb" default="false"/>
    <arg name="use_state_viewer" default="false"/>
	<arg name="object_tracking" default="false" />

    <param name="robot_name" value="$(arg robot_name)"/>
    <param name="robot_hw" value="vrep"/>

    <include if="$(eval arg('robot_name') == 'skkurobot')" file="$(find skkurobot_description)/launch/default.launch">
        <arg name ="sim" value="$(arg sim)"/>
        <arg name="sim_env" default="$(arg sim_env)"/>
    </include>
    <include if="$(eval arg('robot_name') == 'social_robot')" file="$(find social_robot_description)/launch/default.launch">
        <arg name ="sim" value="$(arg sim)"/>
        <arg name="sim_env" value="$(arg sim_env)"/>
        <arg name="sync_mode" value="$(arg sync_mode)"/>
    </include>

    <!-- socialrobot_task_planner -->
    <include file="$(find socialrobot_task)/launch/planning_system.launch"/>

    <!-- socialrobot_interface -->
    <include file="$(find socialrobot_interface)/launch/interface_manager.launch">
       <arg name ="use_mongodb" value="$(arg use_mongodb)"/>
       <arg name ="object_tracking" value="$(arg object_tracking)"/>
       <arg name ="use_knowledge" value="$(arg use_knowledge)"/>
       <arg name ="use_reasoner" value="$(arg use_reasoner)"/>
    </include>

    <!-- socialrobot_behavior -->
    <include if="$(arg use_behavior)" file="$(find socialrobot_behavior)/launch/behavior.launch"/>

    <!-- socialrobot_motion -->
    <include file="$(find socialrobot_motion)/launch/motion_plan.launch"/>

    <!-- socialrobot_knowledge -->
    <include if="$(arg use_knowledge)" file="$(find context_manager)/launch/mkm.launch">
       <arg name ="use_knowledge" value="$(arg use_knowledge)"/>
    </include>

    <!-- socialrobot_reasoner -->
    <include if="$(arg use_reasoner)" file="$(find ir_repositioning)/launch/reasoner.launch"/>

    <!-- socialrobot_perception -->
    <include if="$(arg sim)" file="$(find perception_interface)/launch/vrep_perception.launch"/>
    <include unless="$(arg sim)" file="$(find perception_interface)/launch/perception.launch"/> 

	<!-- state machine -->
	<include file="$(find socialrobot_interface)/launch/state_machine.launch">
       <arg name ="use_state_viewer" value="$(arg use_state_viewer)"/>
    </include>
</launch>
